doc/api/n-api.md

```diff
@@ -0,0 +1,3150 @@
+- `[out] byte_offset`: The byte offset within the data buffer from which
+to start projecting the TypedArray
+
+##### Return value
+- `napi_ok` if the API succeeded.
+
+##### Description
+This API returns various properties of a typed array. 
+Warning: Use caution while using this API since the underlying data buffer
+is managed by the VM
+
+#### *napi_get_value_bool*
+
+##### Signature
+```C
+napi_status napi_get_value_bool(napi_env env, napi_value value, bool* result)
+```
+
+##### Parameters
+- `[in]  env`: The environment that the API is invoked under
+- `[in]  value`: `napi_value` representing JavaScript Boolean
+- `[out] result`: C++ boolean primitive equivalent of the given JavaScript
+Boolean
+
+##### Return value
+- `napi_ok` if the API succeeded.
+- `napi_boolean_expected` if a non-boolean `napi_value` is passed in
+
+##### Description
+This API retuns C++ boolean primitive equivalent of the given JavaScript
+Boolea
+
+#### *napi_get_value_double*
+
+##### Signature
+```C
+napi_status napi_get_value_double(napi_env env,
+                                  napi_value value,
+                                  double* result)
+```
+
+##### Parameters
+- `[in]  env`: The environment that the API is invoked under
+- `[in]  value`: `napi_value` representing JavaScript Number
+- `[out] result`: C++ double primitive equivalent of the given JavaScript
+Number
+
+##### Return value
+- `napi_ok` if the API succeeded.
+- `napi_number_expected` if a non-number `napi_value` is passed in
+
+##### Description
+This API returns the C++ double primitive equivalent of the given JavaScript
+Number.
+
+
+#### *napi_get_value_external*
+
+##### Signature
+```C
+napi_status napi_get_value_external(napi_env env,
+                                    napi_value value,
+                                    void** result)
+```
+
+##### Parameters
+- `[in]  env`: The environment that the API is invoked under
+- `[in]  value`: `napi_value` representing JavaScript External value
+- `[out] result`: Pointer to the data wrapped by the JavaScript External value
+
+##### Return value
+- `napi_ok` if the API succeeded.
+- `napi_invalid_arg` if a non-external `napi_value` is passed in
+
+##### Description
+This API returns the Pointer to the data wrapped by the JavaScript
+External value
+
+#### *napi_get_value_int32*
+
+##### Signature
+```C
+napi_status napi_get_value_int32(napi_env env,
+                                 napi_value value,
+                                 int32_t* result)
+```
+
+##### Parameters
+- `[in]  env`: The environment that the API is invoked under
+- `[in]  value`: `napi_value` representing JavaScript Number
+- `[out] result`: C++ int32 primitive equivalent of the given JavaScript Number
+
+##### Return value
+- `napi_ok` if the API succeeded.
+- `napi_number_expected` if a non-number `napi_value` is passed in
+
+##### Description
+This API returns the C++ int32 primitive equivalent
+of the given JavaScript Number
```

Comment:
These should probably mention what happens when the value does not fit into the corresponding range?